sql/wsrep_binlog.cc

```diff
@@ -434,44 +430,40 @@ void wsrep_wait_for_turn_in_group_commit(THD *thd) {
     } else {
       WSREP_DEBUG(
           "Thread with id (%d) waiting for its turns in wsrep group"
-          " commit queue",
-          thd->thread_id());
+          " commit queue - waiting for (%d)",
+          thd->thread_id(),
+          wsrep_group_commit_queue.front()->thread_id());
       mysql_cond_wait(&COND_wsrep_group_commit, &LOCK_wsrep_group_commit);
     }
   }
 
-  mysql_mutex_unlock(&LOCK_wsrep_group_commit);
-  DBUG_VOID_RETURN;
+  return;
 }
 
 void wsrep_unregister_from_group_commit(THD *thd) {
-  DBUG_ENTER("wsrep_unregister_from_group_commit");
+  DBUG_TRACE;
   if (wsrep_emulate_bin_log || thd == NULL) {
     /* Binlog is off, no need to maintain group commit queue */
-    DBUG_VOID_RETURN;
-  }
-
-  /* thd can be NULL if the transaction is being committed
-  during recovery using XID. */
-  if (!thd) {
-    DBUG_VOID_RETURN;
+    /* thd can be NULL if the transaction is being committed
+       during recovery using XID. */
+    return;
   }
 
-  mysql_mutex_lock(&LOCK_wsrep_group_commit);
+  MUTEX_LOCK(guard, &LOCK_wsrep_group_commit);
 
   if (!thd->wsrep_enforce_group_commit) {
     /* Said handler was not register for wsrep group commit */
     mysql_mutex_unlock(&LOCK_wsrep_group_commit);
```

Comment:
This `mysql_mutex_unlock(&LOCK_wsrep_group_commit);` also needs to be removed as this is no longer needed.

Reply:
Fixed in bef8fc0d66d2d8caf6caed762cafb3dd16cbe9b6